packages/squiggle-lang/src/rescript/FR/FR_Danger_Aggregate.res

```diff
@@ -0,0 +1,171 @@
+          Belt.Int.toString(minLength) ++ " elements",
+        )
+      | false => {
+          let checkedIndividualXs = E.A.fmap(x =>
+            switch (x, x > 1.0, x < 0.0) {
+            | (0.0, _, _) => Error("0 is not a probability")
+            | (1.0, _, _) => Error("1 is not a probability")
+            | (_, true, _) => Error("Probabilities can't be higher than 1")
+            | (_, _, true) => Error("Probabilities can't be lower than 0")
+            | (_, false, false) => Ok(x)
+            }
+          , xs)
+          let checkedCollectiveXs = E.A.R.firstErrorOrOpen(checkedIndividualXs)
+          let result = switch checkedCollectiveXs {
+          | Ok(xs) => fn(xs)
+          | Error(e) => Error(e)
+          }
+          result
+        }
+      }
+    }
+    let probabilityToOdds = p => p /. (1.0 -. p)
+    let oddsToProbability = o => o /. (1.0 +. o)
+    let sum = xs => E.A.reduce(xs, 0.0, (a, b) => a +. b)
+    let mean = xs => sum(xs) /. Belt.Float.fromInt(E.A.length(xs))
+
+    let geomMean = (xs: array<float>) => {
+      let xsLogs = E.A.fmap(x => Js.Math.log2(x), xs)
+      let meanXsLogs = mean(xsLogs)
+      let answer = Js.Math.pow_float(~base=2.0, ~exp=meanXsLogs)
+      answer
+    }
+    let arithmeticMean = (xs: array<float>) => {
+      let sumXs = E.A.reduce(xs, 0.0, (a, b) => a +. b)
+      let meanXs = sumXs /. Belt.Float.fromInt(E.A.length(xs))
+      meanXs
+    }
+
+    let geomMeanOfOdds = xs => {
+      let arrayOfOdds = E.A.fmap(p => probabilityToOdds(p), xs)
+      let arrayOfLogsOfOdds = E.A.fmap(p => Js.Math.log2(p), arrayOfOdds)
+      let meanOfLogsOfodds = mean(arrayOfLogsOfOdds)
+      let geomMeanOfOdds = Js.Math.pow_float(~base=2.0, ~exp=meanOfLogsOfodds)
+      let result = oddsToProbability(geomMeanOfOdds)
+      result
+    }
+
+    let extremizedGeometricMeanOfOdds = (~xs, ~extremizationParameter=1.5, ()) => {
+      let arrayOfOdds = E.A.fmap(p => probabilityToOdds(p), xs)
+      let arrayOfLogsOfOdds = E.A.fmap(p => Js.Math.log2(p), arrayOfOdds)
+      let extremizedSumOfLogsOfOdds = extremizationParameter *. mean(arrayOfLogsOfOdds)
+      let extremizedGeomMeanOfOdds = Js.Math.pow_float(~base=2.0, ~exp=extremizedSumOfLogsOfOdds)
+      let result = oddsToProbability(extremizedGeomMeanOfOdds)
+      result
+    }
+
+    let neyman = xs => {
+      let n = Belt.Int.toFloat(E.A.length(xs))
+      let d =
+        n *.
+        (Js.Math.sqrt(3.0 *. Js.Math.pow_float(~base=n, ~exp=2.0) -. 3.0 *. n +. 1.0) -. 2.0) /.
+        (Js.Math.pow_float(~base=n, ~exp=2.0) -. n -. 1.0)
+      let result = extremizedGeometricMeanOfOdds(~xs, ~extremizationParameter=d, ())
+      result
+    }
+
+    let samotsvety = xs => {
+      let sortedXs = Belt.SortArray.stableSortBy(xs, (a, b) => a > b ? 1 : -1)
+      let middleXs = Belt.Array.slice(sortedXs, ~offset=1, ~len=E.A.length(xs) - 2)
+      Js.log2("MiddleXs: ", middleXs)
+      let answer = geomMeanOfOdds(middleXs)
```

Comment:
Do "samotsvety" is the geoMeanOdds of the items, with the highest and lowest removed? Seems a bit hacky. Fine to have in Danger, but not sure how much I'd want to promote.

Reply:
Added this as an alias instead.